### PR TITLE
Update NVIDIA GPU operator componentry

### LINKED ISSUE #1907: GPU tests failing consistently in testgrid
/kind bug

The GPU-enabled cluster tests [have been failing](https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capz-periodic-e2e-full-main) in k8s testgrid. Need to find the root cause.

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

The GPU-enabled test spec began failing a couple weeks ago. This updates the GPU operator components to recently released versions, which seems to fix what was essentially a runtime C code compilation error inside the nvidia-driver-daemonset pod.

**Which issue(s) this PR fixes**:

Fixes #1907

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mboersma** [comment]
I made this a work-in-progress because I'm adding some test logging that would have clarified the error.

File: templates/flavors/nvidia-gpu/gpu-operator-components.yaml [hunk 1]
```diff
@@ -344,7 +344,7 @@ spec:
         - name: node-feature-discovery-master
           securityContext:
             {}
-          image: "quay.io/kubernetes_incubator/node-feature-discovery:v0.6.0"
+          image: "k8s.gcr.io/nfd/node-feature-discovery:v0.9.0"
```
Comment (CecileRobertMichon) [inline templates/flavors/nvidia-gpu/gpu-operator-components.yaml:347]: this is a good example of one we could switch to using the remote spec directly in the future #1917

**Comment by CecileRobertMichon** [comment]
@mboersma planning on cutting v1.1 today, should we try to include this in there?

**Comment by mboersma** [comment]
> should we try to include this...?

That would be great. I'm validating some changes locally (now that we have power again) and then I'll push them out to the PR.

**Comment by CecileRobertMichon** [comment]
```
capz-e2e: Workload cluster creation Creating a GPU-enabled cluster with a single control plane node and 1 node | 21m43s
-- | --
```

The full test suite failed with an ipv6 and an AKS failure but the GPU test passed, which is what we're looking for here. Going to look into the two failures but in the interest of time and being able to include this in the release:

/lgtm
/approve

**Pull request merged**